src/decide.ts

```diff
@@ -166,18 +194,36 @@ async function assessCandidateHelper(
 	if (hashesToExclude.includes(candidateMeta.infoHash)) {
 		return { decision: Decision.INFO_HASH_ALREADY_EXISTS };
 	}
-	const sizeMatch = compareFileTreesIgnoringNames(candidateMeta, searchee);
-	if (!sizeMatch) {
-		return { decision: Decision.SIZE_MISMATCH };
+
+	if (matchMode === MatchMode.PARTIAL) {
+		const partialMatch = compareFileTreesForPartialMatch(
+			candidateMeta,
+			searchee,
+			minPartialMatch
+		);
+		if (!partialMatch) {
+			return { decision: Decision.SIZE_MISMATCH };
+		}
+	} else {
+		const sizeMatch = compareFileTreesIgnoringNames(candidateMeta, searchee);
+		if (!sizeMatch) {
+			return { decision: Decision.SIZE_MISMATCH };
+		}
```

Comment:
if you look at the rest of this function, it's entirely early returns with no nesting. Can we find a way to flatten this logic?